internal/scanner/scanner.go

```diff
@@ -109,6 +113,67 @@ func (s *Scanner) ScanNamespace(nsName string) error {
 	return s.reportStore.SaveAll()
 }
 
+func (s *Scanner) ScanClusterWideResources() error {
+	log.Info().Msg("cluster wide scan started")
+	policies, skippedNum, err := s.policiesFetcher.GetClusterAdmissionPolicies()
+	if err != nil {
+		return err
+	}
+	log.Debug().
+		Dict("dict", zerolog.Dict().
+			Int("policies to evaluate", len(policies)).
+			Int("policies skipped", skippedNum),
+		).Msg("cluster admission policies count")
+	auditableResources, err := s.resourcesFetcher.GetClusterWideResourcesForPolicies(context.Background(), policies)
+	if err != nil {
+		return err
+	}
+	// create PolicyReport
+	clusterReport := report.NewClusterPolicyReport("cluster-report")
+	clusterReport.Summary.Skip = skippedNum
+	// Iterate through all auditableResources. Each item contains a list of resources and the policies that would need
+	// to evaluate them.
+	for _, resource := range auditableResources {
+		for _, policy := range resource.Policies {
+			url, err := s.resourcesFetcher.GetPolicyServerURLRunningPolicy(context.Background(), policy)
+			if err != nil {
+				// TODO what's the better thing to do here?
+				log.Error().Err(err)
+				continue
+			}
+			for _, resource := range resource.Resources {
+				admissionRequest := resources.GenerateAdmissionReview(resource)
+				auditResponse, responseErr := sendAdmissionReviewToPolicyServer(url, admissionRequest, &s.httpClient)
+				if responseErr != nil {
+					// log error, will end in PolicyReportResult too
+					log.Error().Err(responseErr).Dict("response", zerolog.Dict().
+						Str("admissionRequest name", admissionRequest.Request.Name).Str("policy", policy.GetName()).Str("resource", resource.GetName()),
+					).
+						Msg("error sending AdmissionReview to PolicyServer")
+				} else {
+					log.Debug().Dict("response", zerolog.Dict().
+						Str("uid", string(auditResponse.Response.UID)).Bool("allowed", auditResponse.Response.Allowed).Str("policy", policy.GetName()).Str("resource", resource.GetName()),
+					).
+						Msg("audit review response")
+					clusterReport.AddResult(policy, resource, auditResponse, responseErr)
+				}
+			}
+		}
+	}
+	err = s.reportStore.AddClusterPolicyReport(&clusterReport)
```

Comment:
maybe this should be `s.reportStore.UpdateClusterPolicyReport()`. Right now both `AddClusterPolicyReport()` and `UpdateClusterPolicyReport()` are the same, but that doesn't mean it is the case in the future. Also, if the ClusterPolicyReport already exists because of a prior scan, we should be updating it.